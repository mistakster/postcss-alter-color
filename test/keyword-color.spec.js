const setup = require('./utils/setup');

it('should alter simple color in a single property', () => {
  const source = `
div {
  color: black
}
`;

  return setup(source, {from: 'black', to: 'red'})
    .then(css => expect(css).toMatchSnapshot());
});

it('should alter simple color in a single property and donâ€™t touch others', () => {
  const source = `
div {
  color: black;
  border: none;
  width: 100%;
  height: 50px;
}
`;

  return setup(source, {from: 'black', to: 'red'})
    .then(css => expect(css).toMatchSnapshot());
});

it('should alter color in a complex value', () => {
  const source = `
div {
  color: black;
  border: 1px solid white;
  outline: 1px solid black
}
`;

  return setup(source, {from: 'black', to: 'red'})
    .then(css => expect(css).toMatchSnapshot());
});

it('should alter simple color in multiple properties', () => {
  const source = `
div {
  color: black;
  background: white
}
`;

  return setup(source, {from: 'black', to: 'red'})
    .then(css => expect(css).toMatchSnapshot());
});

it('should alter simple color to hex color a single property', () => {
  const source =`
div {
  color: black
}
`;

  return setup(source, {from: 'black', to: '#fffeee'})
    .then(css => expect(css).toMatchSnapshot());
});

it('should alter simple color to short hex color a single property', () => {
  const source = `
div {
  color: black
}
`;

  return setup(source, {from: 'black', to: '#79b'})
    .then(css => expect(css).toMatchSnapshot());
});
