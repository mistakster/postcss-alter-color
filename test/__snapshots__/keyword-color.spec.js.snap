// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should alter a color keyword in a single property 1`] = `
"
div {
  color: red
}
"
`;

exports[`should alter a color keyword in a single property and donâ€™t touch others 1`] = `
"
div {
  color: red;
  border: none;
  width: 100%;
  height: 50px;
}
"
`;

exports[`should alter a color keyword in the complex values 1`] = `
"
div {
  color: red;
  border: 1px solid white;
  outline: 1px solid red
}
"
`;

exports[`should replace a color keyword with hex color 1`] = `
"
div {
  color: #fffeee
}
"
`;

exports[`should replace a color keyword with hsl() function 1`] = `
"
div {
  color: hsl(30,67%,17%);
}
"
`;

exports[`should replace a color keyword with hsla() function 1`] = `
"
div {
  color: hsla(30,67%,17%,0.5);
}
"
`;

exports[`should replace a color keyword with rgb() function 1`] = `
"
div {
  color: rgb(55,55,55);
}
"
`;

exports[`should replace a color keyword with rgba() function 1`] = `
"
div {
  color: rgba(55,55,55,0.5);
}
"
`;

exports[`should replace a color keyword with short hex color 1`] = `
"
div {
  color: #79b
}
"
`;
